test/mocks/StakingRewardsMock.sol

```diff
@@ -165,11 +165,15 @@ contract StakingRewardsMock {
         emit RewardAdded(reward);
     }
 
-    function setRewardsDuration(uint256 _rewardsDuration) external onlyOwner {
-        require(
-            block.timestamp > periodFinish,
-            "Previous rewards period must be complete before changing the duration for the new period"
-        );
+    function setRewardsDuration(uint256 _rewardsDuration) external onlyOwner updateReward(address(0)) {
+        uint256 periodFinish_ = periodFinish;
+        if (block.timestamp < periodFinish_) {
+            uint256 leftover = (periodFinish_ - block.timestamp) * rewardRate;
+            rewardRate = leftover / _rewardsDuration;
+            lastUpdateTime = block.timestamp;
```

Comment:
Are you sure we need this? 
Inside `updateReward` we have `lastUpdateTime = lastTimeRewardApplicable();` and in `lastTimeRewardApplicable` we have:
`return block.timestamp < periodFinish ? block.timestamp : periodFinish;`

Reply:
Indeed it's not necessary, I only kept it to keep the logic more similar with `notifyRewardAmount()` so that it's more immediately obvious that the code in that `if` is basically just doing `notifyRewardAmount(0);`. But thinking about it now I think it's probably cleaner to just remove the line.